src/app/components/search-suggestions/search-suggestions.component.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  Component,
+  OnInit,
+  Input,
+  EventEmitter,
+  Output,
+  HostListener,
+  OnChanges
+} from "@angular/core";
+import { GetDataService } from "../../services/get-data/get-data.service";
+
+@Component({
+  selector: "app-search-suggestions",
+  templateUrl: "./search-suggestions.component.html",
+  styleUrls: ["./search-suggestions.component.scss"]
+})
+export class SearchSuggestionsComponent implements OnInit, OnChanges {
+  @Input() searchParam: string;
+  @Output() setShowSuggestions = new EventEmitter<boolean>();
+  @Output() setSuggestionSearchParam = new EventEmitter<string>();
+
+  public showSuggestions = false;
+  public loading = false;
+  public searchSuggestions: {};
+  public suggestionsCategory = "completions";
```

Comment:
Is it worth defining an enum-like type for this? It would only give us static checks, which may not be worth much, but actually might catch some attempts to set this to an invalid value. IIRC, you can use the actual string literals as a type, like `"completions" | "suggestions"`.

Reply:
I have now implemented an enum as the search suggestion category in my latest commit. It would be great to get some feedback on this as I'm not used to using enums in javascript.